From d10974dca2e2cd10f8aaa8205f46fd9689591dee Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 05 Sep 2022 14:06:49 +0300
Subject: [PATCH]  [SW-102303] gaudi3: pcie: remove disable hot reset hw sm

In earlier platforms, disabling of hot reset handling by HW state
machine was done as part of pcie link up sequence. This configuration
is not required for gaudi3, so skip it.

Change-Id: I65af433329dd2d8e84d3bf44bf956351d23311d6
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 2142a7e..0e86612 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -287,6 +287,11 @@
 	regmap_writel(0x0, mmD0_PCIE_PHY_PHYX_MPLL_SHORT_LOCK_EN_3);
 }
 
+void pcie_disable_hot_rst_hw_sm(void)
+{
+	/* No configuration required in Gaudi3 */
+}
+
 uint8_t get_sris_mode_from_strap(void)
 {
 	uint32_t boot_strap;
diff --git a/common/pcie.c b/common/pcie.c
index f637da9..20c211d 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -77,6 +77,12 @@
 __weak void pcie_phy_set_pll_short_lock(void) {}
 __weak void pcie_wa_snps_dcc(void) {}
 
+__weak void pcie_disable_hot_rst_hw_sm(void)
+{
+	/* disable HOT reset handling by HW state machine */
+	regmap_writel(CPU_HOT_RST_CTRL_MASK, mmCPU_HOT_RST);
+}
+
 void pcie_config(void)
 {
 	uint32_t link_ctrl;
@@ -145,9 +151,7 @@
 		/* enable flr handling by HW state machine */
 		enable_flr_handling_fsm();
 
-		/* disable HOT reset handling by HW state machine */
-		regmap_writel(CPU_HOT_RST_CTRL_MASK,
-			      mmCPU_HOT_RST);
+		pcie_disable_hot_rst_hw_sm();
 
 		regmap_writel(0x3, mmLTSSM_EN);
 	} else {
